components/tenant-fetcher/internal/tenant/converter.go

```diff
@@ -14,21 +16,33 @@ func NewConverter() *converter {
 func (c *converter) ToEntity(in model.TenantModel) tenant.Entity {
 	return tenant.Entity{
 		ID:             in.ID,
-		Name:           in.TenantId,
+		Name:           in.Name,
 		ExternalTenant: in.TenantId,
-		ProviderName:   in.TenantProvider,
+		Parent:         NewNullString(in.ParentInternal),
+		Type:           in.Type,
+		ProviderName:   in.Provider,
 		Status:         in.Status,
 	}
 }
 
-func (c *converter) FromEntity(in *tenant.Entity) *model.TenantModel {
-	if in == nil {
-		return nil
-	}
-	return &model.TenantModel{
+func (c *converter) FromEntity(in tenant.Entity) model.TenantModel {
+	return model.TenantModel{
 		ID:             in.ID,
+		Name:           in.Name,
 		TenantId:       in.ExternalTenant,
-		TenantProvider: in.ProviderName,
+		ParentInternal: in.Parent.String,
```

Comment:
The same check for sql.NullString validity and assign either the value or the "".

Reply:
It think it would be `""` as default if it's `NULL` in the DB because the `in.Parent.String` is a `string` object, not a `*string`.